kolibri_explore_plugin/collectionviews.py

```diff
@@ -103,26 +105,28 @@ def get_latest_channels(self):
         }
 
     def get_extra_channel_ids(self):
-        all_channel_ids = _get_channel_ids_for_all_content_manifests(
-            self.language
+        all_channel_ids = _get_channel_ids_for_all_collections(
+            self.state.language_id
         )
         return all_channel_ids.difference(self.get_channel_ids())
 
     def get_latest_extra_channels(self):
         """Return set of extra channel id and latest version tuples"""
-        all_channels = _get_latest_channels_for_all_content_manifests(
-            self.language
+        all_channels = _get_latest_channels_for_all_collections(
+            self.state.language_id
         )
         return all_channels.difference(self.get_latest_channels())
 
     def read_from_static_collection(
-        self, grade, name, language, validate=False
+        self, name, sequence, language_id, validate=False
     ):
-        self.grade = grade
-        self.name = name
-        self.language = language
+
+        self.state, created = CollectionState.objects.get_or_create(
+            name=name, sequence=int(sequence)
+        )
```

Comment:
It looks like (if we move this below `super().read(manifest_filename, validate)`) we could do…

```suggestion
        data = self._manifest_data.get("metadata", {})
        self.state, created = CollectionState.objects.get_or_create(
            name=name, sequence=int(sequence), defaults={
                "title": data.get("title"),
                "subtitle": data.get("subtitle"),
                "description": data.get("description"),
                "required_gigabytes": int(data.get("required_gigabytes", 0)),
            }
        )
```

And then we could drop that whole `if created` block.